### PR TITLE
Rebuild stream chunks to openAI object

### LINKED ISSUE #383: [Feature]: Helper function to rebuild streaming + function calling object
### The Feature

If you want a complete object after doing a streamed function call, it can get pretty complicated 

there's 3 streaming chunk structures
- first chunk
- mid chunk (after first, before last streamed chunk)
- last chunk 

So you need to find a way to merge the 3 formats into 1 unified chunk. 

Look at test_streaming for output examples - https://github.com/BerriAI/litellm/blob/15bc5f2bdc96923394e75e6e4a9c8f81b8078313/litellm/tests/test_streaming.py#L441

It would be helpful to expose a helper function to let people rebuild their stream objects 

possible solution:
```
from litellm import stream_chunk_builder 

response = completion(..., stream=true) 
chunks = []
for chunk in response: 
  chunks.append(chunk)

print(stream_chunk_builder(chunks))
```

### Motivation, pitch

Needed it for final output validation testing

### Twitter / LinkedIn details

_No response_

### PR DESCRIPTION
**Pull request by WilliamEspegren**
Fixes #383 

Rebuilds the stream chunks to the openAI object except it doesn't include the "usage".

### CODE REVIEW

**Comment by ghost** [comment]
Hey @WilliamEspegren can you test it with this: 
```
curl --location 'https://api.openai.com/v1/chat/completions' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer openai-api-key' \
--data '{
  "model": "gpt-3.5-turbo-0613",
  "messages": [
    {"role": "user", "content": "What is the weather like in Boston?"}
  ],

  "functions": [
    {
      "name": "get_current_weather",
      "description": "Get the current weather in a given location",
      "parameters": {
        "type": "object",
        "properties": {
          "location": {
            "type": "string",
            "description": "The city and state, e.g. San Francisco, CA"
          },
          "unit": {
            "type": "string",
            "enum": ["celsius", "fahrenheit"]
          }
        },
        "required": ["location"]
      }
    }
  ]
}'
```


i believe current implementation would fail for function calling + Streaming

**Comment by WilliamEspegren** [comment]
That is true, let me add that ability. Great that you catched it!

**Comment by WilliamEspegren** [comment]
@krrishdholakia How do you setup function calling with liteLLM.completion? Is it function_call=True, functions=[list_of_functions]?

There doesn't seem to be any tests written with function calls.

**Comment by WilliamEspegren** [comment]
Hey @krrishdholakia, my latest commit added support for function_calling 👍

**Comment by ghost** [comment]
hey @WilliamEspegren for the function calling+streaming scenario, the test is in test_streaming.py. 

It seems like your branch has some merge conflicts, are you able to resolve them? it's the only blocker for merge

**Comment by WilliamEspegren** [comment]
@krrishdholakia The merge conflicts are now solved 👍

**Comment by ghost** [comment]
merged 👍

**Comment by ghost** [comment]
Do you have a twitter / linkedin? would love to give you a shoutout when we announce the new feature

**Comment by WilliamEspegren** [comment]
Yes @krrishdholakia. My LinkedIn is https://www.linkedin.com/in/william-espegren/

😀

File: litellm/__init__.py [hunk 1]
```diff
@@ -78,7 +78,7 @@ def identify(event_details):
 ####### Secret Manager #####################
 secret_manager_client = None
 ####### COMPLETION MODELS ###################
-open_ai_chat_completion_models = [
+open_ai_chat_completion_models: str = [
```
Comment (ishaan-jaff) [inline litellm/__init__.py:81]: @WilliamEspegren fyi this failed our linting - this is a list

Comment (WilliamEspegren) [reply litellm/__init__.py:81]: 😬ooopsie, great catch by you @ishaan-jaff

**Pull request merged**